plugins/inputs/prometheus/kubernetes.go

```diff
@@ -135,6 +144,31 @@ func podReady(statuss []*corev1.ContainerStatus) bool {
 	return true
 }
 
+func podSelector(p *Prometheus) ([]k8s.Option, error) {
+	options := []k8s.Option{}
+
+	var err error
+
+	if len(p.KubernetesLabelSelector) > 0 {
+		_, err1 := labels.Parse(p.KubernetesLabelSelector)
```

Comment:
This is nice, I didn't think we would be able check the syntax ahead of time.  What could be nice is to check this in an Init() function, so that we can report syntax errors on startup.  Over in `prometheus.go`, add a new function:
```go
func (p *Prometheus) Init() error {
    // check selectors
}
```

This function gets called on plugin startup once, if you return an error, then Telegraf will refuse to start.

Reply:
@danielnelson : thank you, will do so